### PR TITLE
[quantization] Refactor CLE , add algorithm doc and tests

### PR DESCRIPTION
**Pull request by zk1998**

### CODE REVIEW

File: examples/quantization/qat.py [hunk 1]
```diff
@@ -26,6 +26,25 @@ def main_worker(args):
...
+        # For per-tensor quantization, if there are many outliers in the weight, CLE can significantly improve the
+        # quantization accuracy
+        if args.cle:
+            from tinynn.graph.quantization.algorithm.cross_layer_equalization import cross_layer_equalize
+
+            model = cross_layer_equalize(model, dummy_input, device)
+
+        # If your model do not have BatchNorm, for example, in the RepVGG-style deploy model,
+        # their BN fused to conv when doing model reparameter. We provide 'model_restore_bn' to restore BatchNorm.
+        if args.bn_restore:
```
Comment (zk1998) [inline examples/quantization/qat.py:43]: Not sure when this option should be enabled.

File: tests/cross_layer_equalization_test.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import unittest
```
Comment (zk1998) [inline tests/cross_layer_equalization_test.py:1]: Add to `.github / unittest.yml`.

File: tinynn/graph/quantization/algorithm/algorithm_zh-CN.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+需要进行CLE的原模型
+* dummy_input:
+
+模型示例输入，用于进行TinyNN的计算图生成
+* device (torch.device):
+
+当前使用的设备
+* threshold (float):
+
+不进行CLE的权重范围阈值，对于给定的通道，使用```scale=sqrt(abs_max(weight1_cn)/abs_max(weight2_cn))```计算缩放系数scale，
```
Comment (zk1998) [inline tinynn/graph/quantization/algorithm/algorithm_zh-CN.md:31]:
```suggestion
不进行CLE的权重范围阈值。对于给定的通道，使用```scale=sqrt(abs_max(weight1_cn)/abs_max(weight2_cn))```计算缩放系数scale，
```

File: tinynn/graph/quantization/algorithm/algorithm_zh-CN.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+默认设置为0.5。
+* work_dir (str):
+
+CLE算法中间结果的保存目录
+* cle_iters  (int):
+
+进行CLE的次数，CLE次数越多，权重分布将更加平滑，但是实验结果来看，cle_iters并非越大越好，当前默认值为2，推荐1~5次CLE即可。
+* hba_flag (bool):
+
+是否进行HBA操作，用于在带有BN的模型进行CLE之后，消除由CLE引入的bias放大的问题，默认为False关闭。
```
Comment (zk1998) [inline tinynn/graph/quantization/algorithm/algorithm_zh-CN.md:43]: HBA?

File: tinynn/graph/quantization/algorithm/algorithm_zh-CN.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+1. [x] 跨层均衡算法([cross layer equalization](https://arxiv.org/abs/1906.04721))
+2. [ ] 偏置修正算法([bias correction](https://arxiv.org/abs/1906.04721))
+3. [ ] 激活值平滑算法([smooth quant](https://arxiv.org/abs/2211.10438))
+
+## 跨层均衡算法（Corss Layer Equalization, CLE）
+
+在某些模型中，其输出通道之间的权重分布存在极大差异，而对于per-tensor量化而言，使用单独一个量化参数来量化所有权重，会产生显著的量化误差。例如，如果一个通道的权重分布在[-128,128],而另一个通道分布在[-0.5,0.5]之间，那么在INT8量化时，后者将被全部量化为0
+我们称之为这类权重存在"离群点现象"，尤其是在大量使用DW卷积和重参数方法的模型中(例如MobileOne)，离群点现象更加显著。
+
+我们根据[Qualcomm](https://arxiv.org/abs/1906.04721)提出的跨层均衡算法，在TinyNeuralNetwork中集成了CLE算法：
```
Comment (zk1998) [inline tinynn/graph/quantization/algorithm/algorithm_zh-CN.md:12]:
```suggestion
我们根据[Qualcomm](https://arxiv.org/abs/1906.04721)提出的跨层均衡算法，在代码中集成了CLE算法：
```

File: tinynn/graph/quantization/algorithm/algorithm_zh-CN.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+# 量化算法：
+[English](algorithm.md)
+1. [x] 跨层均衡算法([cross layer equalization](https://arxiv.org/abs/1906.04721))
```
Comment (zk1998) [inline tinynn/graph/quantization/algorithm/algorithm_zh-CN.md:3]: 大写

File: tinynn/graph/quantization/algorithm/algorithm_zh-CN.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
+# 量化算法：
+[English](algorithm.md)
+1. [x] 跨层均衡算法([Cross Layer Equalization](https://arxiv.org/abs/1906.04721))
+2. [ ] 偏置修正算法([Bias Correction](https://arxiv.org/abs/1906.04721))
+3. [ ] 激活值平滑算法([Smooth Quant](https://arxiv.org/abs/2211.10438))
+
+## 跨层均衡算法（Corss Layer Equalization, CLE）
```
Comment (zk1998) [inline tinynn/graph/quantization/algorithm/algorithm_zh-CN.md:7]:
```suggestion
## 跨层均衡算法（Cross Layer Equalization, CLE）
```

**Review by peterjc123 — APPROVED** [review]

**Pull request merged**